client/state/themes/actions.js

```diff
@@ -194,39 +195,79 @@ export function receiveThemes( data, site, queryParams, responseTime ) {
 	};
 }
 
-export function activate( theme, site, source = 'unknown' ) {
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been triggered
+ *
+ * @param  {String}  themeId Theme to be activated
+ * @param  {Object}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivation( themeId, siteId ) {
+	return {
+		type: THEME_ACTIVATE_REQUEST,
+		themeId,
+		siteId,
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been successfull
+ *
+ * @param  {Object}  theme Theme received
+ * @param  {Number}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivated( theme, siteId ) {
```

Comment:
> I think that they should be called `themeActivationRequestAction`, `themeActivationRequestSuccessAction`, `themeActivationRequestFailureAction` maybe a bit long but exactly describe what is going on.

That's too long 🙀 . By convention, we should use verbs (not nouns) for actions (`activate`, not `activation`), and no `Action` suffices.

> We would loose symmetry - all other actions have actions creators, and introduce exception in pattern how we handle actions

Symmetry is in fact my motivation too, but wrt to the naming of the action creators corresponding to the `THEME_ACTIVATE_REQUEST_SUCCESS` and `THEME_ACTIVATE_REQUEST_FAILURE` action name constants.

OTOT, consistency with `activateTheme` would be nice (more on that function below).

> We loose ability to name what we are doing - this is my thing where I like self describing code, dislike inlining

I see your point -- I think it's a bit mitigated by the (quite explanatory) action name constants in the `type` fields.

> I also think that in the end `activateTheme` should not be in actions. Its just a procedure that dispatches actions.

Wut? 😄 Where else would you want it to live? (My take: If it's something that we need to `dispatch` -- and it definitely is -- it's an action and should live in an `actions.js` file.)

I'll give the naming some more thought.